Fastlane folder to fastlanecore

chreno opened the pull request:
First part of Pull Request #5825.
Moved fastlane_folder to fastlane_core.

chreno committed:
Moved fastlane_folder to fastlane_core

chreno committed:
Using FastlaneCore as module

mfurtak commented:
Follow-up plan is to remove all but `available_lanes` from `Fastlane::FastlaneFolder`, right?

mfurtak commented:
This looks good. We will have to do a _fastlane_core_ release once this is merged, before we can get the follow-up _fastlane_ side of PR happily depending on it and building correctly 👍

chreno commented:
@mfurtak Thanks! As follow-up plan I moved available_lanes to cli_tools_distributor.rb, since this is the only place it is used. Also deleted fastlane_folder from fastlane.
